cmd/query/app/server.go

```diff
@@ -131,7 +123,67 @@ func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.
 	healthServer.SetServingStatus("jaeger.api_v3.QueryService", grpc_health_v1.HealthCheckResponse_SERVING)
 
 	grpc_health_v1.RegisterHealthServer(server, healthServer)
-	return server, nil
+}
+
+func createGRPCServer(
+	ctx context.Context,
+	legacy bool,
+	querySvc *querysvc.QueryService,
+	metricsQuerySvc querysvc.MetricsQueryService,
+	options *QueryOptions,
+	tm *tenancy.Manager,
+	telset telemetery.Setting,
+) (*grpc.Server, error) {
+	var grpcServer *grpc.Server
+	if legacy {
+		var grpcOpts []grpc.ServerOption
```

Comment:
I would prefer we minimize the code movements. You had a function
```
createGRPCServer
  // create server
  // register endpoints
```

you can refactor it into
```
createGRPCServerLegacy
  // create server

registerEndpoints
  // register endpoints
```
This ^ would be very minimal code change, just a few lines. Then add new function for OTEL gRPC, and then add a dispatching function `createGRPCServer`